tests/unit/core/server/testServer.js

```diff
@@ -83,9 +111,68 @@ describe('core/server/base', () => {
       );
 
       const response = await testClient({ stubRoutes })
-        .get('/en-US/firefox/simulation-of-a-non-existent-page').end();
+        .get('/en-US/firefox/simulation-of-a-non-existent-page')
+        .end();
```

Comment:
I had an idea for making this better. What do you think of this?

```js
const response = await testClient({ stubRoutes })
  .get('/en-US/firefox/simulation-of-a-non-existent-page');
shudownServer(response);
```

`shudownServer()` would just call `response.end()` but it would make it really clear what's happening. Also, we can link to https://github.com/visionmedia/supertest/issues/100 in a comment.

Reply:
I dig it.

Reply:
mmh this is not as easy as what I thought since `await` resolves the promise. I do not get a `supertest` instance but a response (with `req` and `res`) that is the same with or without `end()`. In other words, I cannot call `.end()` on `response`.

Reply:
A solution could be to transform the `testClient()` function into something like `createApp()`, then have a call to `supertest(app)` to get the test client, and later shutdown the server:

```
const app = createApp(...);
const client = supertest(app);

// ...

app.close();
```

WDYT? Slighty more verbose, but explicit.